### ISSUE TITLE
Screen get's stuck after few minutes in printing status menu

### ISSUE THREAD

**Issue by mrcg** [mcuadros/OctoPrint-TFT#34]
I really love the OctoPrint-TFT addon.
Unfortunately i have the issue that OctoPrint-TFT get's stuck after few minutes.

The problem happens only in the printing status menu.
In all other menu's the issue is not happening.
That's the reason why i think it is not related to screen timeout settings.

The issue happens also in the printing status menu if printer is in idle mode (no print ongoing).
After few minutes screen get's stuck (no screen updates) and it is also not possible to go back to home screen.

Would be great to get this fixed soon. Thanks for support.

**Comment by jofegibo** [comment]
In my case,

I see it get's stuck on printing status menu even after printing have finished.

**Comment by ppamidimarri** [comment]
I have the same issue. 

I love this UI on my Pi touchscreen, best way by far to use OctoPi when I am home!

**Comment by EissaAl** [comment]
same issue

**Comment by EissaAl** [comment]
problem which is the screen goes blank after 10 minutes if i touch it, it will come back unless if i left it for 1 hour or more when i touch it it will stay blank.
Linux octopi 4.14.34-v7+
i fixed this problem by using 
`sudo rpi-update`
now the Linux octopi is at 4.14.58-v7+
and when i touch the screen it will come back :D

**Comment by EissaAl** [comment]
omg just fixed the issue by following this https://askubuntu.com/questions/476036/xset-unable-to-open-display

then this https://raspberrypi.stackexchange.com/questions/752/how-do-i-prevent-the-screen-from-going-blank


what i did to solve this problem 
i tried this command `xset s noblank` but it keeps saying xsetroot:  unable to open display ':0'
this mean my display is at DISPLAY=:1 so i entered `export DISPLAY=:1`
then 
`xset s off         # don't activate screensaver`
`xset -dpms         # disable DPMS (Energy Star) features.`
`xset s noblank     # don't blank the video device`

**Comment by mrcg** [comment]
Hey EissaAl,
i would like to be happy with you.
But the issue here is a total other issue than yours.
For your mentioned issue there was already solution posted in the closed topic section.

The topic here is only happening in the "printing status" menu after some time.
And also the screen does not become blank.

So for the moment there seems to be no solution for this issue.

**Comment by zombu2** [comment]
Same problem here after a couple minutes on printing status the display goes unresponsive and does not update

**Comment by mrcg** [comment]
Yesterday i have reproduced the issue another two times and the failure scenario is always the same.
On "GET /api/job" there is no response from the server visible abd then octoprint-tft gets stuck.
I also can see that response from server can take up to 5 seconds which could also cause some issues as the polling interval in octoprint-tft is also 5 seconds.

Suggestion to solve the issue could be:
1. Increase polling time to 10s.
2. Add robustness to the code so that missing HTTP reply does not cause any issue. E.g. Store previous variables until the get overwritten with udpated one.
3. Remove remaining printing time calculation and use instead the remaining printing time variable from the octoprint server response.

Unfortunately i have not much coding experience on souce code like this. Can someone help here?

**Comment by mrcg** [comment]
I am stopping my investigations here because i could not understand how to modify the "status.go" source.
:-(

**Comment by The5py** [comment]
All working well in my end now the only issue keep happening to me seem to be known issues but making it not possible to use octoprint-tft:
Screen get's stuck after few minutes in printing status menu (issue #34)
 and  
Freeze while printing (issue #28)
Any solution for them yet?
Is there anyone it's not happening to him?

**Comment by ppamidimarri** [comment]
Thank you mrcg for your investigations. I tried to test your theory, so here are the changes I made to status.go:

```
        switch s.Progress.Completion {
        case 100:
//              text = fmt.Sprintf("Job Completed in %s", time.Duration(int64(s.Job.LastPrintTime)*1e9))
                text = fmt.Sprintf("Job Completed")
        case 0:
                text = "Warming up ..."
        default:
//              e := time.Duration(int64(s.Progress.PrintTime) * 1e9)
//              l := time.Duration(int64(s.Progress.PrintTimeLeft) * 1e9)
//              text = fmt.Sprintf("Elapsed/Left: %s / %s", e, l)
                text = fmt.Sprintf("Job in progress")
//              if l == 0 {
//                      text = fmt.Sprintf("Elapsed: %s", e)
//              }
        }

        m.left.Label.SetLabel(text)

```
Attached is the deb package that I built for Raspbian Stretch (OctoPi image) with these changes. My printer is broken right now (bed thermistor needs changing) so I can't test it right now. If you want to test it, install this file with dpkg -i and check if the problem is fixed. 
[octoprint-tft_0.0-git04dd1f0-1_armhf.zip](https://github.com/mcuadros/OctoPrint-TFT/files/2316669/octoprint-tft_0.0-git04dd1f0-1_armhf.zip)

**Comment by mrcg** [comment]
@ppamidimarri Many thanks for compiling the deb package!!! I will be able to test it earliest next week on monday. For trouble shooting it is good that you have taken out the printing time estimation completely. This should allow us to prove the concept in order to define final solution. I will report after testing on my 3d printer next week. Stay tuned!

**Comment by ppamidimarri** [comment]
@mrcg I fixed my printer and ran one job. I can see the new text "Job in progress" instead of the times. But unfortunately the screen got stuck around 30% into a one-hour job. So this doesn't seem to be the real issue. I will try to look further.

**Comment by mrcg** [comment]
@ppamidimarri Good that you have fixed your printer and have been able to test.

I think we need to modify also this:

` m.pb.SetFraction(s.Progress.Completion / 100)`

Beside the calculation it would be great if you could make one deb package by only increasing the new background task time from 5 to 10 seconds. This should (hopefully) help to give more time for the reply if timeout is the real root cause:

` m.b = NewBackgroundTask(time.Second*5, m.update)`

WDYT?

**Comment by ppamidimarri** [comment]
Yes, I changed that frequency of API call to once every 10 seconds. 
```
m.b = NewBackgroundTask(time.Second*10, m.update)
```
I reverted the rest of the changes in status.go to original, except this line for clearer logic:
```
        default:
                e := time.Duration(int64(s.Progress.PrintTime) * 1e9)
                l := time.Duration(int64(s.Progress.PrintTimeLeft) * 1e9)
                if l == 0 {
                        text = fmt.Sprintf("Elapsed: %s", e)
                } else {
                        text = fmt.Sprintf("Elapsed/Left: %s / %s", e, l)
                }
```
Here is the Debian package with these changes. I will test this now. 
[octoprint-tft_0.0_git04dd1f0-1_armhf.zip](https://github.com/mcuadros/OctoPrint-TFT/files/2319551/octoprint-tft_0.0_git04dd1f0-1_armhf.zip)

**Comment by mrcg** [comment]
I have no access to my printer until monday.
I am crossing my fingers that your tests are successful.

**Comment by ppamidimarri** [comment]
I tried to diagnose the problem using more log messages at the debug level in status.go in the code @mrcg highlighted. I also added this line to debian/octoprint-tft.service to see if any error messages are generated when the status panel hangs:

```
StandardError=journal
```
However I can't see any error messages using journalctl -t xinit around the time of the status panel hangs. Debug messages added to the line updating the text label show up in the journal until just before the panel hangs. 

I am unable to fix the real problem. So I decided to put in a small workaround that reduces the frequency of this problem. In ui.go, there is a line that opens the status panel automatically when the printer is working on a job. I commented this out. 
```
//                      ui.Add(StatusPanel(ui, DefaultPanel(ui)))
```
Attached is the deb file with this tweak. With this, the only time OctoPrint-TFT hangs is if you go into the status panel manually and leave it there. You just have to remember to back out of it to the main panel if you don't want it to hang. 
[octoprint-tft_0.0_git04dd1f0-1_armhf.zip](https://github.com/mcuadros/OctoPrint-TFT/files/2322023/octoprint-tft_0.0_git04dd1f0-1_armhf.zip)

**Comment by EissaAl** [comment]
not only in the status panel, if you left it without connecting the printer it will freeze after some time.
https://i.imgur.com/zjM0VJl.jpg

**Comment by The5py** [comment]
@ppamidimarri For now with the lack of fix for this bug or any solution I think your workaround is good and I am going to use it.
Thought about another little nice to have to add, on the event of status panel opening I suggest to invoke a timer which will get out of the status panel into the main screen automatically after a short period of time by that to help the user which forgot to get out of this screen not to get stuck in the middle of printing which is annoying...

BTW anyone knows about a command to restart the octoprint-tft process in case it is stucked already during print without the need to reboot the entire raspberry pie

**Comment by mrcg** [comment]
@ppamidimarri  Will try your workaround but anyhow we should try to find the root cause from status.go code. Question: a) The timer increase did not solve the issue? b) Do you know if the timer change (10s) was really taking effekt to allow more time for receiving the feedback?

Another question: Is it possible to explain shortly how to create own deb package after modification of the source files? Or is this to complex to explain in few steps?

My idea would be to almost clear the status.go and then add successively one more function in order to find the guilty code. 

How can we find out if missing answer from the server could cause the issues? Because - as already sayed above - in my wireshark traces on localhost i can see several times that there is not response from the server on the "GET /api/job" request.

**Comment by ppamidimarri** [comment]
@EissaAl, yes the Octoprint-TFT UI will stay on the splash screen unless your OctoPi is connected to your printer. See this other issue: #15. I don't understand the code or the go language enough to fix that.

**Comment by ppamidimarri** [comment]
@The5py, this is a good idea. I will see if I know enough to get this change done. 

One way I know to get Octoprint-TFT to restart without rebooting the Pi is to install the deb package again with sudo dpkg -i. It works most of the time, but I once had the Pi stuck on a completely blank screen instead of the UI.

**Comment by ppamidimarri** [comment]
@mrcg, correct, the timer change doesn't solve the problem. 

I know the timer change was taking effect. I confirmed that by adding in status go:
```
      Logger.Debug(fmt.Sprintf("Received: ", text))
```
just before:
```
      m.left.Label.SetLabel(text)
```
Once I had this running, I checked what the program was logging, using the command `journalctl -t xinit` and I noticed that it was logging a response once every 10 seconds. If I change the timer to 5 seconds, it logs a response every 5 seconds. So the change of timer is definitely happening, but it is not helping prevent the status panel getting stuck. 
Here is a quick summary of how you can build your own deb packages with changes you make to the code. 

1. Create a directory on your Pi where you will make the code changes. There, clone the respository with `git clone github.com/mcuadros/OctoPrint-TFT`.
2. Follow the steps listed by the original developer in the Readme file install from source. Use the command `curl -fsSL get.docker.com -o get-docker.sh` to get the Docker installation script. Then run the script with `sh get-docker.sh`. This step can take a long time. If you get errors, repeat. 
3. Now you can build the deb package with your code changes. Go to the directory where you cloned OctoPrint-TFT and in that directory, use the command `make build`. This will take a while. Once it is done, the deb package will be in the subdirectory build/stretch. There will be two deb files there, use the one with the shorter name, without "dbgsym" in the name. 

Please note that I am really new to Debian, Raspbian, OctoPrint, Docker, Go and GitHub. The last time I did programming was 15 years ago. I learned all these steps by googling each piece. So I may not be able to help if you run into issues with the above. I highly suggest making a backup of your Pi before you start this. Also, read the output of your commands carefully, and if there are errors in one step, don't move to the next before you take care of them. Good luck!

**Comment by EissaAl** [comment]
@ppamidimarriI mean even if u connect the printer it will stay in the splash screen because I didn't connect the printer for some time. same as the status panel problem.

**Comment by ppamidimarri** [comment]
@EissaAl Got it, sounds like there is a bigger problem somewhere that makes the application freeze when the OctoPrint server doesn't respond for a while. I am totally out of my depth now, I hope the original developer has some time to help us out!

**Comment by mrcg** [comment]
@ppamidimarri  Based on your description i was able to compile the deb package on my own. Many thanks for this !!!!

I am trying to take out some code in order to hopefully find the guilty part.
Yesterday i took out the whole "updateJob" function from status.go where i have expected the root cause for the problem .... but unfortunetely it's not the case. The issue still happened which means it's cause by some other code. Today i will take out some more code but due to the fact it's very time consuming the activitiy will be limited.

**Comment by mrcg** [comment]
Some GOOD NEWS !!!

After disabling the "updateTemperature" function from status.go (remove line 145) the issue did not reoccure again after about 3-4h testing. It looks very promissing that the issue is somehow caused by this code:

```
func (m *statusPanel) updateTemperature() {
	s, err := (&octoprint.StateRequest{Exclude: []string{"sd"}}).Do(m.UI.Printer)
	if err != nil {
		Logger.Error(err)
		return
	}

	m.doUpdateState(&s.State)

	m.tool1.Hide()
	for tool, s := range s.Temperature.Current {
		text := fmt.Sprintf("%s: %.0f°C / %.0f°C", strings.Title(tool), s.Actual, s.Target)
		switch tool {
		case "bed":
			m.bed.Label.SetLabel(text)
		case "tool0":
			m.tool0.Label.SetLabel(text)
		case "tool1":
			m.tool1.Label.SetLabel(text)
			m.tool1.Show()
		}
	}
}
```

I am thinking now about this two potential workaround solutions:
1. Remove temperatur information from status screen completely because i have this information already locally available from 3d printer's LCD display.
2. Remove the tool1 handling as both of my 3d printers have only tool0.
3. Developer to fix the real issue in the code. ;-)

**Comment by EissaAl** [comment]
@mrcg how I can modify the code?
sudo nano then what is the location of the file

**Comment by ppamidimarri** [comment]
@mrcg Excellent progress! Removing updateTemperature is a no-brainer for me. I never look at that when I want to know how hot my printer is, I look at the printer's LCD screen. 

I will look into this more tonight and try to test it with several print jobs.

**Comment by The5py** [comment]
@mrcg - It works great!
I remaked as well- updateTemperature on line 145 compiled it and it is up and running for some short prints nothing getting stuck for the moment

Great workaround for now!
I think checking it with only removing the second hotend (tool1 and leave tool0) is a good path to try investigate.

Now looking for a solution or workaround for #18 
@mrcg & @ppamidimarri does the temperature profiles works good for you?

**Comment by EissaAl** [comment]
Can anyone tell me where is that file?

**Comment by The5py** [comment]
OctoPrint-TFT/ui/status.go

**Comment by EissaAl** [comment]
@The5py using ssh
i have to use sudo nano then the file location right?

**Comment by ppamidimarri** [comment]
@The5py I don't use that feature because I slice on my computer. That issue #18 is definitely a bug in OctoPrint's web interface which seems to store some values as ints and some as strings.

**Comment by mrcg** [comment]
Hi all,
removing tool1 from status.go completely did not solve the issue ... unfortunately.
So i think the issue is caused from the 1st section of the updateTemperature function:

```
	s, err := (&octoprint.StateRequest{Exclude: []string{"sd"}}).Do(m.UI.Printer)
	if err != nil {
		Logger.Error(err)
		return
	}

	m.doUpdateState(&s.State)
```
One of my next change will be as discussed in this forum:
https://stackoverflow.com/questions/25356602/golang-panic-crash-prevention

Unfortunately i was not able until now to do this change successfully without compiling error afterwards.
Currently i have ongoing print and cannot proceed in testing new code.

I think the exclude string command could cause the issue because it's not used in any other variable update process. To change/remove this part will be last idea after pending panic crash prenvention. If this does not point to the root cause i will remove the temp update completely (including the icons) from status.go code.

**Comment by ppamidimarri** [comment]
@mrcg I removed the temperature display from status.go in `createMainBox()` by commenting out ` grid.Add(m.createTemperatureBox())` and then commenting out all calls to `defer m.updateTemperature()` (they occur in `createPrintButton()`, `createPauseButton()` and `createStopButton()`) in addition to commenting out the call in `update()` to `m.updateTemperature()`. 

I reverted my old workaround in ui.go of not adding the status panel automatically. 

With these changes, the UI doesn't show the temperatures at all so it looks clean, and the status screen doesn't seem to hang. I am three hours into a four hour print and my OctoPrint-TFT status screen has been up all this while and it is still updating the elapsed and left times. I will test this some more with other prints. 

Thank you for finding a really great workaround for this problem that almost had me uninstalling OctoPrint-TFT!

**Comment by EissaAl** [comment]
@mrcg @ppamidimarri the problem if u connect the printer it will stay in the splash screen after not connecting the printer for some time is not fixed by deleting "updateTemperature" function from status.go (line 145) but everything else is fixed.

**Comment by mrcg** [comment]
Yes, correct. This correction here is only fixing the issue from the printing status menu (status.go).

**As workaround solultion the following lines have been removed from status.go (v0.1.2) in order to prevent OctoPrint-TFT to crash after some time in printing status menu. The workaround removes also the temperature status from bed, tool0 and tool1:**

	( 59) grid.Add(m.createTemperatureBox())
	(103) defer m.updateTemperature()
	(117) defer m.updateTemperature()
	(132) defer m.updateTemperature()
	(145) m.updateTemperature()

Attached the compiled deb packages with the changes mentioned above.
Installation of the deb packages like this:
```
sudo dpkg -i octoprint-tft_v0.1.2_#34-fix_stretch.deb
or
sudo dpkg -i octoprint-tft_v0.1.2_#34-fix_jessie.deb
```

[octoprint-tft_v0.1.2_#34-fix_stretch.zip](https://github.com/mcuadros/OctoPrint-TFT/files/2333378/octoprint-tft_v0.1.2_.34-fix_stretch.zip)

[octoprint-tft_v0.1.2_#34-fix_jessie.zip](https://github.com/mcuadros/OctoPrint-TFT/files/2333383/octoprint-tft_v0.1.2_.34-fix_jessie.zip)

**Comment by ppamidimarri** [comment]
@mrcg I have tested the fix with several jobs over the last day, including a 12 hour job. The problem with status screen hanging is completely solved with your fix. Thanks again for your persistence! 
As an idea to better use the screen real estate now available with the temperature display gone, here is the change I made to show the elapsed and left times separately, along with the ETA (which adds current time + left time) to show when your current job will finish. Here is the code I changed in the `updateJob()` function within status.go, starting on line 226:

```
        default:
                e := time.Duration(int64(s.Progress.PrintTime) * 1e9)
                l := time.Duration(int64(s.Progress.PrintTimeLeft) * 1e9)
                eta := time.Now().Add(l).Format("Mon Jan 2, 3:04:05 PM")
                if l == 0 {
                        text = fmt.Sprintf("Elapsed: %s", e)
                } else {
                        text = fmt.Sprintf("Elapsed: %s, Left: %s, ETA: %s", e,$
                }
```
I also changed the function `filenameEllipsis()` starting at line 241:
```
        if len(name) > 46 {
                return name[:43] + "..."
        }
```

**Comment by ppamidimarri** [comment]
@EissaAl It looks like there is a logging call made in the code in ui.go that verifies the connection state every 5 seconds. Next time you see this problem, please do this:
1. SSH into your OctoPi 
2. Run the command `journalctl -t xinit | grep go`
3. In the output see what the last couple of lines say and may be post it here
This may help one of us diagnose and attempt to fix what is going on.

**Comment by EissaAl** [comment]
@ppamidimarri 
`journalctl -t xinit | grep go
Aug 28 19:56:23 octopi xinit[611]: Kernel command line: 8250.nr_uarts=1 bcm2708_                                                                    fb.fbwidth=640 bcm2708_fb.fbheight=480 bcm2708_fb.fbswap=1 vc_mem.mem_base=0x3ec                                                                    00000 vc_mem.mem_size=0x40000000  dwc_otg.lpm_enable=0 console=tty1 console=ttyS                                                                    0,115200 root=/dev/mmcblk0p2 rootfstype=ext4 elevator=deadline rootwait consoleb                                                                    lank=0 fbcon=map:10 fbcon=font:ProFont6x11 logo.nologo
Aug 28 19:56:29 octopi xinit[611]: time="2018-08-28T19:56:29Z" level=info msg="O                                                                    ctoPrint's config file found: \"/home/pi/.octoprint/config.yaml\"" file=main.go                                                                     func=main.readConfig line=98
Aug 28 19:56:29 octopi xinit[611]: time="2018-08-28T19:56:29Z" level=info msg="U                                                                    sing \"http://localhost:5000\" as server address" file=main.go func=main.init.0                                                                     line=47
Aug 28 19:56:29 octopi xinit[611]: time="2018-08-28T19:56:29Z" level=info msg="F                                                                    ound API key at \"/home/pi/.octoprint/config.yaml\" file" file=main.go func=main                                                                    .init.0 line=55
Aug 28 19:56:30 octopi xinit[611]: time="2018-08-28T19:56:30Z" level=debug msg="                                                                    New background task started" file=common.go func="ui.(*BackgroundTask).Start" li                                                                    ne=109
Aug 28 19:56:30 octopi xinit[611]: time="2018-08-28T19:56:30Z" level=debug msg="                                                                    Unexpected error: Get http://localhost:5000/api/connection: net/http: HTTP/1.x t                                                                    ransport connection broken: malformed MIME header line: Not found" file=ui.go fu                                                                    nc="ui.(*UI).verifyConnection" line=110
Aug 28 19:56:35 octopi xinit[611]: time="2018-08-28T19:56:35Z" level=info msg="C                                                                    onnection offline, connecting: Closed" file=ui.go func="ui.(*UI).verifyConnectio                                                                    n" line=132
Aug 28 19:56:41 octopi xinit[611]: time="2018-08-28T19:56:41Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:56:56 octopi xinit[611]: time="2018-08-28T19:56:56Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:57:01 octopi xinit[611]: time="2018-08-28T19:57:01Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:57:06 octopi xinit[611]: time="2018-08-28T19:57:06Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:57:11 octopi xinit[611]: time="2018-08-28T19:57:11Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:57:16 octopi xinit[611]: time="2018-08-28T19:57:16Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:57:21 octopi xinit[611]: time="2018-08-28T19:57:21Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:57:26 octopi xinit[611]: time="2018-08-28T19:57:26Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:57:31 octopi xinit[611]: time="2018-08-28T19:57:31Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:57:36 octopi xinit[611]: time="2018-08-28T19:57:36Z" level=info msg="C                                                                    onnection offline, connecting: Closed" file=ui.go func="ui.(*UI).verifyConnectio                                                                    n" line=132
Aug 28 19:57:41 octopi xinit[611]: time="2018-08-28T19:57:41Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:57:46 octopi xinit[611]: time="2018-08-28T19:57:46Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:57:51 octopi xinit[611]: time="2018-08-28T19:57:51Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:57:56 octopi xinit[611]: time="2018-08-28T19:57:56Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:58:01 octopi xinit[611]: time="2018-08-28T19:58:01Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:58:06 octopi xinit[611]: time="2018-08-28T19:58:06Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:58:11 octopi xinit[611]: time="2018-08-28T19:58:11Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:58:16 octopi xinit[611]: time="2018-08-28T19:58:16Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:58:21 octopi xinit[611]: time="2018-08-28T19:58:21Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:58:26 octopi xinit[611]: time="2018-08-28T19:58:26Z" level=info msg="C                                                                    onnection offline, connecting: Closed" file=ui.go func="ui.(*UI).verifyConnectio                                                                    n" line=132
Aug 28 19:58:31 octopi xinit[611]: time="2018-08-28T19:58:31Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 28 19:58:36 octopi xinit[611]: time="2018-08-28T19:58:36Z" level=info msg="P                                                                    rinter is ready" file=ui.go func="ui.(*UI).verifyConnection" line=119
Aug 28 19:58:40 octopi xinit[611]: time="2018-08-28T19:58:40Z" level=debug msg="                                                                    New background task started" file=common.go func="ui.(*BackgroundTask).Start" li                                                                    ne=109
Aug 29 02:09:53 octopi xinit[611]: time="2018-08-29T02:09:53Z" level=debug msg="                                                                    Background task closed" file=common.go func="ui.(*BackgroundTask).loop" line=136
Aug 29 02:10:11 octopi xinit[611]: time="2018-08-29T02:10:11Z" level=info msg="C                                                                    onnection offline, connecting: Closed" file=ui.go func="ui.(*UI).verifyConnectio                                                                    n" line=132
Aug 29 02:10:16 octopi xinit[611]: time="2018-08-29T02:10:16Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:10:21 octopi xinit[611]: time="2018-08-29T02:10:21Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:10:26 octopi xinit[611]: time="2018-08-29T02:10:26Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:10:31 octopi xinit[611]: time="2018-08-29T02:10:31Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:10:36 octopi xinit[611]: time="2018-08-29T02:10:36Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:10:41 octopi xinit[611]: time="2018-08-29T02:10:41Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:10:46 octopi xinit[611]: time="2018-08-29T02:10:46Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:10:51 octopi xinit[611]: time="2018-08-29T02:10:51Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:10:56 octopi xinit[611]: time="2018-08-29T02:10:56Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:11:01 octopi xinit[611]: time="2018-08-29T02:11:01Z" level=info msg="C                                                                    onnection offline, connecting: Closed" file=ui.go func="ui.(*UI).verifyConnectio                                                                    n" line=132
Aug 29 02:11:06 octopi xinit[611]: time="2018-08-29T02:11:06Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:11:11 octopi xinit[611]: time="2018-08-29T02:11:11Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:11:16 octopi xinit[611]: time="2018-08-29T02:11:16Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:11:21 octopi xinit[611]: time="2018-08-29T02:11:21Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:11:26 octopi xinit[611]: time="2018-08-29T02:11:26Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:11:31 octopi xinit[611]: time="2018-08-29T02:11:31Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:11:36 octopi xinit[611]: time="2018-08-29T02:11:36Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:11:41 octopi xinit[611]: time="2018-08-29T02:11:41Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:11:46 octopi xinit[611]: time="2018-08-29T02:11:46Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:11:51 octopi xinit[611]: time="2018-08-29T02:11:51Z" level=info msg="C                                                                    onnection offline, connecting: Closed" file=ui.go func="ui.(*UI).verifyConnectio                                                                    n" line=132
Aug 29 02:11:56 octopi xinit[611]: time="2018-08-29T02:11:56Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:12:01 octopi xinit[611]: time="2018-08-29T02:12:01Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:12:06 octopi xinit[611]: time="2018-08-29T02:12:06Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:12:11 octopi xinit[611]: time="2018-08-29T02:12:11Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:12:16 octopi xinit[611]: time="2018-08-29T02:12:16Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:12:21 octopi xinit[611]: time="2018-08-29T02:12:21Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:12:26 octopi xinit[611]: time="2018-08-29T02:12:26Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:12:31 octopi xinit[611]: time="2018-08-29T02:12:31Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:12:36 octopi xinit[611]: time="2018-08-29T02:12:36Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:12:41 octopi xinit[611]: time="2018-08-29T02:12:41Z" level=info msg="C                                                                    onnection offline, connecting: Closed" file=ui.go func="ui.(*UI).verifyConnectio                                                                    n" line=132
Aug 29 02:12:46 octopi xinit[611]: time="2018-08-29T02:12:46Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:12:51 octopi xinit[611]: time="2018-08-29T02:12:51Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:12:56 octopi xinit[611]: time="2018-08-29T02:12:56Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:13:01 octopi xinit[611]: time="2018-08-29T02:13:01Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:13:06 octopi xinit[611]: time="2018-08-29T02:13:06Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:13:11 octopi xinit[611]: time="2018-08-29T02:13:11Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:13:16 octopi xinit[611]: time="2018-08-29T02:13:16Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:13:21 octopi xinit[611]: time="2018-08-29T02:13:21Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:13:26 octopi xinit[611]: time="2018-08-29T02:13:26Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:13:31 octopi xinit[611]: time="2018-08-29T02:13:31Z" level=info msg="C                                                                    onnection offline, connecting: Closed" file=ui.go func="ui.(*UI).verifyConnectio                                                                    n" line=132
Aug 29 02:13:36 octopi xinit[611]: time="2018-08-29T02:13:36Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:13:41 octopi xinit[611]: time="2018-08-29T02:13:41Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:13:46 octopi xinit[611]: time="2018-08-29T02:13:46Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:13:51 octopi xinit[611]: time="2018-08-29T02:13:51Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:13:56 octopi xinit[611]: time="2018-08-29T02:13:56Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:14:01 octopi xinit[611]: time="2018-08-29T02:14:01Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:14:06 octopi xinit[611]: time="2018-08-29T02:14:06Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:14:11 octopi xinit[611]: time="2018-08-29T02:14:11Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:14:16 octopi xinit[611]: time="2018-08-29T02:14:16Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:14:21 octopi xinit[611]: time="2018-08-29T02:14:21Z" level=info msg="C                                                                    onnection offline, connecting: Closed" file=ui.go func="ui.(*UI).verifyConnectio                                                                    n" line=132
Aug 29 02:14:26 octopi xinit[611]: time="2018-08-29T02:14:26Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:14:31 octopi xinit[611]: time="2018-08-29T02:14:31Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:14:36 octopi xinit[611]: time="2018-08-29T02:14:36Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:14:41 octopi xinit[611]: time="2018-08-29T02:14:41Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:14:46 octopi xinit[611]: time="2018-08-29T02:14:46Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:14:51 octopi xinit[611]: time="2018-08-29T02:14:51Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:14:56 octopi xinit[611]: time="2018-08-29T02:14:56Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:15:01 octopi xinit[611]: time="2018-08-29T02:15:01Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:15:06 octopi xinit[611]: time="2018-08-29T02:15:06Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:15:11 octopi xinit[611]: time="2018-08-29T02:15:11Z" level=info msg="C                                                                    onnection offline, connecting: Closed" file=ui.go func="ui.(*UI).verifyConnectio                                                                    n" line=132
Aug 29 02:15:16 octopi xinit[611]: time="2018-08-29T02:15:16Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:15:21 octopi xinit[611]: time="2018-08-29T02:15:21Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:15:26 octopi xinit[611]: time="2018-08-29T02:15:26Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:15:31 octopi xinit[611]: time="2018-08-29T02:15:31Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:15:36 octopi xinit[611]: time="2018-08-29T02:15:36Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:15:41 octopi xinit[611]: time="2018-08-29T02:15:41Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:15:46 octopi xinit[611]: time="2018-08-29T02:15:46Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:15:51 octopi xinit[611]: time="2018-08-29T02:15:51Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:15:56 octopi xinit[611]: time="2018-08-29T02:15:56Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:16:01 octopi xinit[611]: time="2018-08-29T02:16:01Z" level=info msg="C                                                                    onnection offline, connecting: Closed" file=ui.go func="ui.(*UI).verifyConnectio                                                                    n" line=132
Aug 29 02:16:06 octopi xinit[611]: time="2018-08-29T02:16:06Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:16:11 octopi xinit[611]: time="2018-08-29T02:16:11Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:16:16 octopi xinit[611]: time="2018-08-29T02:16:16Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:16:21 octopi xinit[611]: time="2018-08-29T02:16:21Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:16:26 octopi xinit[611]: time="2018-08-29T02:16:26Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:16:31 octopi xinit[611]: time="2018-08-29T02:16:31Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:16:36 octopi xinit[611]: time="2018-08-29T02:16:36Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:16:41 octopi xinit[611]: time="2018-08-29T02:16:41Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:16:46 octopi xinit[611]: time="2018-08-29T02:16:46Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:16:51 octopi xinit[611]: time="2018-08-29T02:16:51Z" level=info msg="C                                                                    onnection offline, connecting: Closed" file=ui.go func="ui.(*UI).verifyConnectio                                                                    n" line=132
Aug 29 02:16:56 octopi xinit[611]: time="2018-08-29T02:16:56Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:17:01 octopi xinit[611]: time="2018-08-29T02:17:01Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137
Aug 29 02:17:06 octopi xinit[611]: time="2018-08-29T02:17:06Z" level=info msg="W                                                                    aiting for connection: Detecting baudrate" file=ui.go func="ui.(*UI).verifyConne                                                                    ction" line=137`

**Comment by ppamidimarri** [comment]
@EissaAl Thanks for posting this. Looks like no error is getting logged when the thing just freezes. It is checking every five seconds for a connection, and then it just stops doing that. Not clear to me what the issue is.

**Comment by mrcg** [comment]
@ppamidimarri To extend the filename lenght is a really great addition. I will add this also to my installation later on. Many thanks again for the good cooperation!

**Comment by mrcg** [comment]
Just one more comment. The "Heat Up" menu (temperature.go) contains exactly the same temperature information than the printing status menu (status.go). While the temperature update function in status.go is sporadically crashing the function in the temperature.go does not have this problem.

Final solution would be to implement the temperature update code from temperature.go into status.go.
The method in updating the temperature information is done in different way. Unfortunately i was not able to understand the go language in a way to do this change. 

This would be the final solution.

**Comment by mcuadros** [comment]
Hey everyone, sorry for the lack of answers about this project, but I am really busy with a bunch of other stuff, plus I didn't use octoprint-tft myself, since was created for being used in a printer I was making but I didn't finish it. 

These days, I got a new 3d printer a HicTop Duale3 and it comes with the standard LCD screen, so looks like I am going to back to this project in the coming weeks. 

I will try to solve this problem ASAP.

BTW I will love to get some pictures of the people using this project.

**Comment by mrcg** [comment]
That's really great new's that you proceed to work on this awesome project.
:-)

**Comment by EissaAl** [comment]
My OctoPrint-TFT Setup :)
[https://i.imgur.com/1MFcJKE.jpg](url)
[https://i.imgur.com/onakkzB.jpg](url)
[https://i.imgur.com/krlG849.jpg](url)
[https://i.imgur.com/JdgY2yT.jpg](url)

**Comment by ppamidimarri** [comment]
> Does
> 
> > Yes, correct. This correction here is only fixing the issue from the printing status menu (status.go).
> > **As workaround solultion the following lines have been removed from status.go (v0.1.2) in order to prevent OctoPrint-TFT to crash after some time in printing status menu. The workaround removes also the temperature status from bed, tool0 and tool1:**
> > ```
> > ( 59) grid.Add(m.createTemperatureBox())
> > (103) defer m.updateTemperature()
> > (117) defer m.updateTemperature()
> > (132) defer m.updateTemperature()
> > (145) m.updateTemperature()
> > ```
> > Attached the compiled deb packages with the changes mentioned above.
> > Installation of the deb packages like this:
> > ```
> > sudo dpkg -i octoprint-tft_v0.1.2_#34-fix_stretch.deb
> > or
> > sudo dpkg -i octoprint-tft_v0.1.2_#34-fix_jessie.deb
> > ```
> > [octoprint-tft_v0.1.2_#34-fix_stretch.zip](https://github.com/mcuadros/OctoPrint-TFT/files/2333378/octoprint-tft_v0.1.2_.34-fix_stretch.zip)
> > [octoprint-tft_v0.1.2_#34-fix_jessie.zip](https://github.com/mcuadros/OctoPrint-TFT/files/2333383/octoprint-tft_v0.1.2_.34-fix_jessie.zip)
> 
> does this compiled code include the TFT fix or am I breaking it again by fixing this bug?
> 
> I love this app but am about ready to bin my screen altogether as I seem to need to learn programming to get it running :(

I think @mrcg's files include the fix, so they should just work without any more changes to be made by you.

**Comment by Dryphter** [comment]
When I install the octoprint-tft_v0.1.2_#34-fix_stretch.zip fix, my TFT screen just shows the terminal / command prompt. ie it doesn't load the X server.  Any ideas?

**Comment by mrcg** [comment]
@Dryphter  
After installation you need to configure octoprint-tft:
1. Modifiy octoprint-tft configuration:
`sudo vi /etc/octoprint-tft-environment`
This two changes need to be done ...
OCTOPRINT_HOST=http://localhost
OCTOPRINT_APIKEY= enter your API key here
2. Modify octoprint-tft service configuration:
`sudo vi /lib/systemd/system/octoprint-tft.services`
You need to change TFT output from 0 to 1:
`ExecStart=/usr/bin/xinit /usr/bin/OctoPrint-TFT -- :1 -nolisten tcp -nocursor`

Afterwards octoprint-tft should appear on the LCD ... hopefully. Hope this helps.

**Comment by Dryphter** [comment]
GOT IT! Odd error.
Worked fine with the original 0.1.2 deb file, but whenever I installed the #34-fix deb file, I would only see the terminal/command prompt.

Tried the above suggestion from @mrcg - no go.

So I ran journalctl -u octoprint-tft and looked through the output.  
![image](https://user-images.githubusercontent.com/46200403/50522162-7f160600-0a97-11e9-87d1-fc662891e4f0.png) 
Fatal error trying to open /home/pi/octoprint/ (this is set in the octoprint-tft-environment file)  No idea why as this same file works with the v0.1.2 build.  

I deleted /home/pi/octoprint from the octoprint-tft-environment config file and left it blank.  Boom! started working.  
--------------------------My Settings in case it helps someone else----------------------------------

----------------------------octoprint-tft-environment------------------------

OCTOPRINT_CONFIG_FILE=

OCTOPRINT_HOST=http://localhost

OCTOPRINT_APIKEY=<MY API KEY>

OCTOPRINT_TFT_STYLE_PATH=/opt/octoprint-tft/styles/default/

OCTOPRINT_TFT_RESOLUTION=

----------------/lib/systemd/system/octoprint-tft.service------------------------
ExecStart=/usr/bin/xinit /usr/bin/OctoPrint-TFT -- :0 -nolisten tcp -nocursor

Now to get the Profiles to stop throwing a json error....

**Comment by nickm324** [comment]
I have version 0.1.2 installed and am having the freezing issue after it has been running for a few minutes. Is there a fix for this yet? Shame to have to switch to TouchUI since this is so much nicer.

**Comment by majorkuso** [comment]
I recently installed the octoprint-tft and freezes occassionally, however I can restart it by issuing the command sudo service octoprint-tft restart. it restarts the tft and doesnt stop the print. I may create a cron job to restart it every 2 minutes or so.

**Comment by Antrax-Reloaded** [comment]
Hi, 
I had the same problem and using the solution package, the error disappears but at startup I have an interface that does not allow me to return to previous menus. For example, if I go to the menu to move my axes, I have six buttons but none to return to the main menu.
